samples/durable-task-sdks/dotnet/Agents/PromptChaining/Client/Program.cs

```diff
@@ -0,0 +1,180 @@
+using Azure.Identity;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Client.AzureManaged;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using AgentChainingSample.Shared.Models;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+// Configure logging
+using ILoggerFactory loggerFactory = LoggerFactory.Create(builder =>
+{
+    builder.AddConsole();
+    builder.SetMinimumLevel(LogLevel.Information);
+});
+
+ILogger<Program> logger = loggerFactory.CreateLogger<Program>();
+logger.LogInformation("Starting Agent Chaining Sample - Content Creation Client");
+
+// Get environment variables for endpoint and taskhub with defaults
+string endpoint = Environment.GetEnvironmentVariable("ENDPOINT") ?? "http://localhost:8080";
+string taskHubName = Environment.GetEnvironmentVariable("TASKHUB") ?? "default";
+
+// Split the endpoint if it contains authentication info
+string hostAddress = endpoint;
+if (endpoint.Contains(';'))
+{
+    hostAddress = endpoint.Split(';')[0];
+}
+
+// Determine if we're connecting to the local emulator
+bool isLocalEmulator = endpoint == "http://localhost:8080";
+
+// Enable HTTP/2 cleartext support for emulator
+if (isLocalEmulator)
+{
+    AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
+}
+
+// Construct a proper connection string with authentication
+string connectionString;
```

Comment:
Could we simplify this code by simply accepting a connection string as an environment variable? There are a lot of assumptions here in this code which won't always be valid. Accepting a connection string would be a more robust way to configure our sample apps.